### DISCUSSION TITLE
How to add noprefixroute flag when adding address?

### DISCUSSION THREAD

**Discussion by tsv1991** [svinota/pyroute2#1149] [Q&A]
Hi,
we use next one construction:
```
ipr = IPRoute()
ipr.link('add', ifname='br8888', kind='bridge', mcast_snooping=0)
ifindex_br = ipr.link_lookup(ifname='br8888')[0]
ipr.addr('add', index=ifindex_br, address="169.254.169.254", mask=24, scope=253)
```

How we can add flag `noprefixroute` when use `ipr.addr`?
Used the `ip` utility we do it like: `ip a add 169.254.169.254/24 dev br8888 noprefixroute`

**Comment by tsv1991** [comment]
I did it like that:
flags=0x200
`ipr.addr('add', index=ifindex_br, address="169.254.169.254", mask=24, scope=253, flags=0x200)`

**Comment by svinota** [comment]
I believe that the bridge options should be prefixed with `br_`:

```
ipr.link('add', ifname='br8888', kind='bridge', br_mcast_snooping=0)
```

Regarding the address there are flags defined in `pyroute2.netlink.rtnl.ifaddrmsg`:

https://github.com/svinota/pyroute2/blob/696d2bfb638c3e035efe095f1aa4c09edbd9479f/pyroute2/netlink/rtnl/ifaddrmsg.py#L15-L27

Also there are static functions to convert flags, like that:
```python
from pyroute2.netlink.rtnl.ifaddrmsg import ifaddrmsg

ipr.addr(
    'add',
    index=ipr.link_lookup(ifname='br8888'),  # index can be provided as an iterable, the [0] item will be used
    address="169.254.169.254",
    prefixlen=24,
    scope=253,
    flags=ifaddrmsg.names2flags(["IFA_F_NOPREFIXROUTE"])
)
```

May be syntax like `flags=["noprefixroute"]` would be simpler; it would be easy to add, I think.

**Reply by tsv1991** [reply]
> May be syntax like flags=["noprefixroute"] would be simpler; it would be easy to add, I think.

I think this looks like more right way, because it seems universal.
